clap_complete/tests/testsuite/dynamic.rs

```diff
@@ -132,6 +132,119 @@ goodbye-world"
     );
 }
 
+#[test]
+fn suggest_argument_value() {
+    let mut cmd = Command::new("dynamic")
+    .arg(
+        clap::Arg::new("input")
+            .long("input")
+            .short('i')
+            .value_hint(clap::ValueHint::FilePath),
+    )
+    .arg(
+        clap::Arg::new("format")
+            .long("format")
+            .short('F')
+            .value_parser(["json", "yaml", "toml"]),
+    )
+    .args_conflicts_with_subcommands(true);
+    
+    {
+        use std::fs::File;
+        use std::path::Path;
+
+        let testdir = snapbox::dir::DirRoot::mutable_temp().unwrap();
+        let testdir_path = testdir.path().unwrap();
+
+        File::create(Path::new(testdir_path).join("a_file")).unwrap();
+        File::create(Path::new(testdir_path).join("b_file")).unwrap();
+        std::fs::create_dir_all(Path::new(testdir_path).join("c_dir")).unwrap();
+        std::fs::create_dir_all(Path::new(testdir_path).join("d_dir")).unwrap();
+        
+        assert_data_eq!(
+            complete!(cmd, "--input [TAB]", current_dir = Some(testdir_path)),
+            snapbox::str![
+                "a_file
+b_file
+c_dir/
+d_dir/"
+            ],
+        );
+
+        assert_data_eq!(
+            complete!(cmd, "-i [TAB]", current_dir = Some(testdir_path)),
+            snapbox::str![
+                "a_file
+b_file
+c_dir/
+d_dir/"
+            ],
+        );
+
+        assert_data_eq!(
+            complete!(cmd, "--input a[TAB]", current_dir = Some(testdir_path)),
+            snapbox::str![
+                "a_file"
+            ],
+        );
+
+        assert_data_eq!(
+            complete!(cmd, "-i b[TAB]", current_dir = Some(testdir_path)),
+            snapbox::str![
+                "b_file"
+            ],
+        );
+    }
+        
+    
+
+    assert_data_eq!(
+        complete!(cmd, "--format [TAB]"),
+        snapbox::str![
+            "json
+yaml
+toml"
+        ],
+    );
```

Comment:
If I run `cargo test` on this commit, will it fail?

Tests should pass on their commits.  Yes, this means this commit should be showing what the dynamic completions currently report.  Then the following commit will show how the behavior changed through the tests.

Reply:
I have run `cargo test --features="unstable-dynamic"` on this commit, it passed.

Reply:
Looks like the commit order changed (or I overlooked it before).

The tests should be added before features or fixes and they should pass in their own commit.  Then the feature or fix should force the tests to change.

Reply:
I will change back.